### PR TITLE
[IMM32] Rewrite ImmSetStatusWindowPos

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Rewrite `imm32!ImmSetStatusWindowPos` function.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -3129,25 +3129,29 @@ BOOL WINAPI ImmSetOpenStatus(HIMC hIMC, BOOL fOpen)
  */
 BOOL WINAPI ImmSetStatusWindowPos(HIMC hIMC, LPPOINT lpptPos)
 {
-    InputContextData *data = get_imc_data(hIMC);
+    LPINPUTCONTEXT pIC;
+    HWND hwnd;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:3133]:
```suggestion
    HWND hWnd;
```

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -3129,25 +3129,29 @@ BOOL WINAPI ImmSetOpenStatus(HIMC hIMC, BOOL fOpen)
...
+    DWORD idImeThread, idThread;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:3134]: usual win32 code practice: `dwThreadId` etc.

File: dll/win32/imm32/imm.c [hunk 3]
```diff
@@ -3129,25 +3129,29 @@ BOOL WINAPI ImmSetOpenStatus(HIMC hIMC, BOOL fOpen)
...
 
-    TRACE("(%p, %p)\n", hIMC, lpptPos);
+    TRACE("ImmSetStatusWindowPos(%p, {%ld, %ld})\n", hIMC, lpptPos->x, lpptPos->y);
 
-    if (!data || !lpptPos)
-    {
-        SetLastError(ERROR_INVALID_HANDLE);
+    idImeThread = Imm32QueryInputContext(hIMC, 1);
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:3138]: Do you know what this "1" is? Could it be the same as the one I've commented about in your other PR?

File: dll/win32/imm32/imm.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: OK. Renamed.

File: dll/win32/imm32/imm.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: OK. Renamed.

File: dll/win32/imm32/imm.c (hunk 3 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: I don't know at all.

File: dll/win32/imm32/imm.c [hunk 4]
```diff
@@ -3129,25 +3129,29 @@ BOOL WINAPI ImmSetOpenStatus(HIMC hIMC, BOOL fOpen)
...
+    HWND hWnd;
+    DWORD dwImeThreadId, dwThreadId;
 
-    TRACE("(%p, %p)\n", hIMC, lpptPos);
+    TRACE("ImmSetStatusWindowPos(%p, {%ld, %ld})\n", hIMC, lpptPos->x, lpptPos->y);
 
-    if (!data || !lpptPos)
-    {
-        SetLastError(ERROR_INVALID_HANDLE);
+    dwImeThreadId = Imm32QueryInputContext(hIMC, 1);
```
Comment (katahiromz) [inline dll/win32/imm32/imm.c:3138]: Hypothesis: `Imm32QueryInputContext`'s 2nd paramter is same as `Imm32QueryWindow`'s 2nd parameter.

File: dll/win32/imm32/imm.c (hunk 3 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Hypothesis: `Imm32QueryInputContext`'s 2nd paramter is same as `Imm32QueryWindow`'s 2nd parameter.

**Pull request merged**